src/PufferVaultMainnet.sol

```diff
@@ -80,20 +105,107 @@ contract PufferVaultMainnet is PufferVault {
         }
 
         uint256 assets = previewRedeem(shares);
+
+        _checkDailyWithdrawalLimits(assets); //@todo figure if it is better to override `maxWithdraw`
+
+        _wrapETH(assets);
+
         // solhint-disable-next-line func-named-parameters
         _withdraw(_msgSender(), receiver, owner, assets, shares);
 
         return assets;
     }
 
-    // slither-disable-next-line dead-code
-    function _wrap(uint256 amount) internal {
-        _WETH.deposit{ value: amount }();
+    /**
+     * @param withdrawalAmount is the assets amount, not shares
+     */
+    function _checkDailyWithdrawalLimits(uint256 withdrawalAmount) internal {
```

Comment:
function name needs to reflect the functionality -- updates state variables. suggestion: checkandUpdate...

Reply:
Changed the logic for `maxDeposit` and `maxRedeem` so that we are compliant with the 4626 standard.